packages/main/src/plugin/cancellation-token.spec.ts

```diff
@@ -36,43 +36,45 @@ beforeAll(() => {
 test('Should return CancellationToken', async () => {
   const tokenSourceId = cancellationTokenRegistry.createCancellationTokenSource();
   const tokenSource = cancellationTokenRegistry.getCancellationTokenSource(tokenSourceId);
-  const token = tokenSource.token;
+  const token = tokenSource?.token;
   expect(token).toBeDefined();
   expect(token instanceof CancellationTokenImpl).toBe(true);
 });
 
 test('Check if token cancel is triggered when tokenSource cancel', async () => {
   const tokenSourceId = cancellationTokenRegistry.createCancellationTokenSource();
   const tokenSource = cancellationTokenRegistry.getCancellationTokenSource(tokenSourceId);
-  const token = tokenSource.token;
-  const tokenCancelMock = vitest.spyOn(token, 'cancel');
-  tokenCancelMock.mockImplementation(() => {});
-  tokenSource.cancel();
+  const token = tokenSource?.token;
+  expect(token instanceof CancellationTokenImpl).toBe(true);
+  const tokenCancelMock = vitest.spyOn(token as CancellationTokenImpl, 'cancel');
+  tokenSource?.cancel();
   expect(tokenCancelMock).toBeCalled();
 });
 
 test('Check if tokenSource cancel is triggered if tokenSource is disposed', async () => {
   const tokenSourceId = cancellationTokenRegistry.createCancellationTokenSource();
   const tokenSource = cancellationTokenRegistry.getCancellationTokenSource(tokenSourceId);
-  const token = tokenSource.token;
-  const tokenSourceCancelMock = vitest.spyOn(tokenSource, 'cancel');
-  tokenSourceCancelMock.mockImplementation(() => {});
-  const tokenDisposeMock = vitest.spyOn(token, 'dispose');
-  tokenDisposeMock.mockImplementation(() => {});
-  tokenSource.dispose(true);
+  const token = tokenSource?.token;
+  expect(token instanceof CancellationTokenImpl).toBe(true);
+  // assert the object is not undefined and then only of type CancellationTokenSource
+  expect(tokenSource).toBeDefined();
```

Comment:
Isn't tokenSource defined because of the previous assert on line 58 ?

Reply:
fixed, moved the line just after getting tokenSource (line 56)